src/sentry/api/endpoints/project_rule_details.py

```diff
@@ -38,23 +49,89 @@
 logger = logging.getLogger(__name__)
 
 
+class ProjectRuleDetailsPutSerializer(serializers.Serializer):
+    actionMatch = serializers.ChoiceField(
+        choices=(
+            ("all", "All conditions must evaluate to true."),
+            ("any", "At least one of the conditions must evaluate to true."),
+            ("none", "All conditions must evaluate to false."),
+        ),
+        required=False,
+        help_text="A string determining which of the conditions need to be true before any filters are evaluated.",
+    )
+    actions = serializers.ListField(
+        child=RuleNodeField(type="action/event"),
+        required=False,
+        help_text="A list of actions that take place when all required conditions and filters for the rule are met. See [Create an Issue Alert Rule](/api/events/create-an-issue-alert-rule-for-a-project) for valid actions.",
+    )
+    conditions = serializers.ListField(
+        child=RuleNodeField(type="condition/event"),
+        required=False,
+        help_text="A list of triggers that determine when the rule fires. See [Create an Issue Alert Rule](/api/events/create-an-issue-alert-rule-for-a-project) for valid conditions.",
+    )
+    environment = serializers.CharField(
+        required=False, allow_null=True, help_text="The name of the environment to filter by."
+    )
+    filterMatch = serializers.ChoiceField(
+        choices=(
+            ("all", "All filters must evaluate to true."),
+            ("any", "At least one of the filters must evaluate to true."),
+            ("none", "All filters must evaluate to false."),
+        ),
+        required=False,
+        help_text="A string determining which filters need to be true before any actions take place.",
+    )
+    filters = serializers.ListField(
+        child=RuleNodeField(type="filter/event"),
+        required=False,
+        help_text="A list of filters that determine if a rule fires after the necessary conditions have been met. See [Create an Issue Alert Rule](/api/events/create-an-issue-alert-rule-for-a-project) for valid filters.",
+    )
+    frequency = serializers.IntegerField(
+        min_value=5,
+        max_value=60 * 24 * 30,
+        required=False,
+        help_text="How often to perform the actions once for an issue, in minutes.",
+    )
+    name = serializers.CharField(max_length=64, required=False, help_text="The name for the rule.")
```

Comment:
This field should probably go at the top, same for the POST. Sorry, I gave you bad info when I said everything had to be alphabetical, it's prob better to put these params in order of how you'd do it in the UI, with the identifier (in this case name) at the top

Reply:
ok, in that case are we still ordering by required first then by UI or are we just ordering by the order in the UI?

Reply:
It would still be required first, then by ordering in the UI or whatever is most important